docs/spring-boot/core-starter.md

```diff
@@ -0,0 +1,257 @@
+The Symphony BDK for Java provides a _Starter_ module that aims to ease bot developments within a 
+[Spring Boot](https://spring.io/projects/spring-boot) application. 
+
+## Features
+- Configure bot environment through `application.yaml`
+- Subscribe to Real Time Events from anywhere
+- Provide injectable services
+- Ease activities creation
+- Provide `@Slash` annotation to register a [slash command](https://javadoc.io/doc/com.symphony.platformsolutions/symphony-bdk-core/latest/com/symphony/bdk/core/activity/command/SlashCommand.html)
+
+## Installation
+
+The following listing shows the `pom.xml` file that is created when you choose Maven:
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
+    <artifactId>bdk-core-spring-boot</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>bdk-core-spring-boot</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.symphony.platformsolutions</groupId>
+            <artifactId>symphony-bdk-core-spring-boot-starter</artifactId>
+            <version>1.3.0.BETA</version>
+        </dependency>
+    </dependencies>
+</project>
+```
+The following listing shows the `build.gradle` file that is created when you choose Gradle:
+```groovy
+plugins {
+    id 'org.springframework.boot' version "2.3.3.RELEASE"
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter:1.3.0.BETA'
+}
+```
+
+## Create a Simple Bot Application
+As a first step, you have to initialize your bot environment through the Spring Boot `src/main/resources/application.yaml` file: 
+```yaml
+bdk:
+    host: acme.symphony.com
+    bot:
+      username: bot-username
+      privateKeyPath: /path/to/rsa/privatekey.pem
+``` 
+> You can notice here that the `bdk` property inherits from the [`BdkConfig`](https://javadoc.io/doc/com.symphony.platformsolutions/symphony-bdk-core/latest/com/symphony/bdk/core/config/model/BdkConfig.html) class.
+
+As required by Spring Boot, you have to create an `src/main/java/com/example/bot/BotApplication.java` class:
+```java
+@SpringBootApplication
+public class BotApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+}
+```
+
+Now you can create a component for a simple bot application, as the following listing (from `src/main/java/com/example/bot/HelloBot.java`) 
+shows:
+```java
+@Component
+public class HelloBot {
+
+  @Autowired
+  private MessageService messageService;
+
+  @EventListener
+  public void onMessageSent(RealTimeEvent<V4MessageSent> event) {
+    this.messageService.send(event.getSource().getMessage().getStream(),"<messageML>Hello!</messageML>");
```

Comment:
a space is missing between the parameters (formatting psycho....)

out of the scope here but:
event.getSource().getMessage().getStream() is a bit ugly

this.messageService.reply(event, "<messageML>Hello!</messageML>"); would be nice :)

Reply:
> this.messageService.reply(event, "Hello!"); would be nice :)

The problem here is that `RealTimeEvent<T>` is only part of the Spring module. But we could however however assume that `RealTimeEvent` has a `streamId` property, that is common for a Real Time Events. 

Let me have a look!

Reply:
But that would be impossible, as `V4ConnectionRequested` and `V4ConnectionAccepted` events are not related to a stream.

Reply:
FYI, the `ConnectionRequest` and `ConnectionAccept` event, they don't have `streamId` property.

Reply:
or maybe `this.messageService.reply(event.getSource(), "Hello!");` then?